sycl/test/basic_tests/accessor/host_acc_opt.cpp

```diff
@@ -1,20 +1,16 @@
-// RUN: %clangxx -O2 -std=c++17 -I %sycl_include/sycl -I %sycl_include -S -emit-llvm %s -o - | FileCheck %s
+// RUN: %clangxx -O2 -isystem %sycl_include/sycl -isystem %sycl_include -S -emit-llvm %s -o - | FileCheck %s
 
 // The test verifies that the accessor::operator[] implementation is
 // good enough for compiler to optimize away calls to getOffset and
-// getMemoryRange and vectorize the loop.
+// getMemoryRange.
 
 #include <sycl/sycl.hpp>
 
 // CHECK: define {{.*}}foo{{.*}} {
-// CHECK-NOT: call
-// CHECK-NOT: invoke
-// CHECK: load <4 x i32>
-// CHECK-NOT: call
-// CHECK-NOT: invoke
-// CHECK: store <4 x i32>
-// CHECK-NOT: call
-// CHECK-NOT: invoke
+// CHECK-NOT: call{{.*}}getOffset
+// CHECK-NOT: invoke{{.*}}getOffset
+// CHECK-NOT: call{{.*}}getMemoryRange
+// CHECK-NOT: invoke{{.*}}getMemoryRange
```

Comment:
These checks look very fragile. One way to pass this test with call not being optimize - rename internal accessor member functions.
It seems to me that current version checks that optimizer removes all call and invoke instructions from `foo` function. Can we continue using these checks?
Alternative way to improve the test is to check that foo calls `getOffset` and `getMemoryRange` functions before optimizations.

Reply:
One of possible optimization which we can end up having here is a call to some "optimized" function, like `memcpy` that why the check that there are no calls won't work.
I can modify `Acc[I] = Src[I];` to `Acc[I] += Src[I];` to avoid generation of `memcpy`, but it doesn't sound robust as well.

Reply:
Okay.